com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/ShaderPass/ShaderPassLightTransport.hlsl

```diff
@@ -99,18 +101,19 @@ float4 Frag(PackedVaryingsToPS packedInput) : SV_Target
     // We use unity_MetaFragmentControl to make the distinction.
     float4 res = float4(0.0, 0.0, 0.0, 1.0);
 
-    if (unity_MetaFragmentControl.x)
-    {
-        // Apply diffuseColor Boost from LightmapSettings.
-        // put abs here to silent a warning, no cost, no impact as color is assume to be positive.
-        res.rgb = clamp(pow(abs(lightTransportData.diffuseColor), saturate(unity_OneOverOutputBoost)), 0, unity_MaxOutputValue);
-    }
-
-    if (unity_MetaFragmentControl.y)
-    {
-        // emissive use HDR format
-        res.rgb = lightTransportData.emissiveColor;
-    }
+    UnityMetaInput metaInput;
+    metaInput.Albedo = lightTransportData.diffuseColor.rgb;
+    metaInput.Emission = lightTransportData.emissiveColor;
+#if defined(UNLIT_CS_HLSL) || defined(HAIR_CS_HLSL)
```

Comment:
oh sorry I missed this change in my first review. It is forbidden to access surfaceData in any of ShaderPassXXX.hlsl  it will not compile. Each Material can implement what they want.

What you need is to move the code sampling the Material here to each of the function GetLightTransportData() in Material it has been implemented for this.

What you need to do is to add SpecularColor in LightTransportData struct and move the code 
metaInput.SpecularColor = surfaceData.specularColor.rgb in each of the Material (each material have a different way to define spec color so you need to adapt for each material (you can check of the debug mode to output spec color convert the property for each material).

Then in this current files you will have only this:
UnityMetaInput metaInput;
    metaInput.Albedo = lightTransportData.diffuseColor.rgb;
    metaInput.Emission = lightTransportData.emissiveColor;
    metaInput.SpecularColor = lightTransportData.specularColor.rgb;
#endif
#ifdef EDITOR_VISUALIZATION
    metaInput.VizUV = input.texCoord1.xy;
    metaInput.LightCoord = float4(input.texCoord2.xy, input.texCoord3.xy);
#endif

Reply:
Ah actuaully, this SpecularColor was only used in validation debug views that are unsupported by SRPs (likely because their lighting models do not match builtin lighting models). So I just removed SpecularColor from MetaInput in fac9bab